Library/MapleBacon/MapleBacon/ImageGifExtension.swift

```diff
@@ -7,35 +7,35 @@ import ImageIO
 
 extension UIImage {
 
-    class func imageWithCachedData(data: NSData) -> UIImage? {
-        guard data.length > 0 else { return nil }
+    class func image(withCachedData data: Data) -> UIImage? {
+        guard data.count > 0 else { return nil }
         
-        return isAnimatedImage(data) ? animatedImageWithData(data) : UIImage(data: data)
+        return isAnimatedImage(data: data) ? animatedImageWithData(data: data) : UIImage(data: data as Data)
     }
 
-    private class func animatedImageWithData(data: NSData) -> UIImage? {
-        let source = CGImageSourceCreateWithData(data as CFDataRef, nil)
-        return UIImage.animatedImageWithSource(source)
+    private class func animatedImageWithData(data: Data) -> UIImage? {
+        let source = CGImageSourceCreateWithData(data as CFData, nil)
+        return UIImage.animatedImageWithSource(source: source)
     }
 
-    private class func isAnimatedImage(data: NSData) -> Bool {
+    private class func isAnimatedImage(data: Data) -> Bool {
         var length = UInt16(0)
-        data.getBytes(&length, range: NSRange(location: 0, length: 2))
+        (data as NSData).getBytes(&length, range: NSRange(location: 0, length: 2))
```

Comment:
Data is not entirely compatible to NSData but is actually an typealias. so you can downcast ist to NSData to use some of the not existing methods like getBytes